src/cached_object_store.rs

```diff
@@ -81,8 +120,21 @@ impl CachedObjectStore {
                         },
                     )
                     .await?;
+
                 let meta = result.meta.clone();
-                self.save_result(result).await.ok();
+                let mut ghost_guard = self.ghost.lock().await;
+                // ghost queue hit, so the element is not accessed for the first time, we actually write it to
+                // the local cache otherwise we just record it in the ghost queue.
+                if ghost_guard.need_cache(location) {
+                    // TODO(asukamilet): process `save_result` failure, maybe we should put
+                    // identifier to the tail of ghost queue if the disk is full?
+                    self.save_result(result).await.ok();
+                    self.db_stats.object_store_cache_write_num.inc();
+                    ghost_guard.remove(location);
+                } else {
+                    ghost_guard.record(location.clone(), result.meta);
```

Comment:
`record` increases the size of the cache. However, first access doesn't actually cache. Shouldn't we only increase size after we indeed store the file?